app/Http/Controllers/CityController.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\City;
+use Illuminate\Http\Request;
+
+class CityController extends Controller
+{
+    public function index(){
+
+        return view('cities.show',[
+            'links' => City::orderby('id','asc')->select('*')->paginate()->links()
+        ]);
+    }
+
+    public function store(Request $request){
+
+        $attributes = $request->validate([
+            'name' => 'required|unique:cities'
+        ]);
+
+        $city = City::create($attributes);
+        return Response()->json($city);
+    }
+
+    public function getCities(){
+        $cities = City::orderby('id','asc')->select('*')->paginate();
+        $response['data'] = $cities;
+        return response()->json($response);
+    }
+
+    public function update(Request $request, $id){
```

Comment:
```suggestion
    public function update(Request $request, $id): JsonResponse
    {
```
Don't forget about the return type. Also, use a Form Request for validations. Instead of passing the `$id` as a parameter and then performing a `find`, use [route model binding](https://laravel.com/docs/10.x/routing#route-model-binding), which is a built-in feature that does this automatically.